src/main/java/entity/Car.java

```diff
@@ -1,27 +1,32 @@
 package entity;
 
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 public class Car {
     private int id;
-    private static final Logger LOGGER = LogManager.getLogger(Car.class);
+    private Semaphore semaphore;
+    private AtomicBoolean isFree = new AtomicBoolean(true);
 
-    public Car(int id) {
+    public Car(int id, final Semaphore semaphore) {
         this.id = id;
+        this.semaphore = semaphore;
     }
 
     public int getId() {
         return id;
     }
 
-    public void occupy(final Customer customer) {
-        LOGGER.info("Order for client with ID " + customer.getId() + " is being executed");
-        customer.setTripIsDone(true);
+    public AtomicBoolean checkAvailability() {
```

Comment:
You getters/setters should have adjective. Try to think what is true in case when you put something like this:
`if(car.checkAvailability()) {...} `

Construction `if(car.isAvailable()) {..}` is more readable, or not?